src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
+    private static final Object[] empty_arraydata = {};
+
+
+    private T[] array = (T[]) new Object[def_capacity];
+    private int size;
+    private int cursor;
+
+
+    private String errorMsg(int index) {
```

Comment:
```suggestion
    private String getIndexErrorMessage(int index) {
```
Use informative names for your variables and methods.